src/picongpu/include/simulationControl/MySimulation.hpp

```diff
@@ -281,12 +273,24 @@ class MySimulation : public SimulationHelper<simDim>
     virtual void init()
     {
         namespace nvmem = PMacc::nvidia::memory;
+
+        DataConnector &dc = Environment<>::get().DataConnector();
+
         // create simulation data such as fields and particles
-        fieldB = new FieldB(*cellDescription);
-        fieldE = new FieldE(*cellDescription);
-        fieldJ = new FieldJ(*cellDescription);
+        auto fieldB = new FieldB( *cellDescription );
```

Comment:
could do, but `share()` requires a `std::shared_ptr< ISimulationData >` not a `std::shared_ptr<FieldB>`.

`ISimulationData` as no `init()` interface which still need sto be called and I would still need to cast via `std::static_pointer_cast< ISimulationData>` before passing to `share()`.